src/app/api/auth/[...nextauth]/sw360OauthOption.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) Siemens AG, 2024. Part of the SW360 Frontend Project.
+
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+
+// SPDX-License-Identifier: EPL-2.0
+// License-Filename: LICENSE
+
+import { SW360User } from '../../../../../nextauth'
+import crypto from 'crypto';
+import { SW360_API_URL, SW360_REST_CLIENT_ID, SW360_REST_CLIENT_SECRET } from '@/utils/env';
+import { NextAuthOptions } from 'next-auth';
+
+let codeVerifier: crypto.BinaryLike;
+
+const sw360OauthOption: NextAuthOptions = {
+  providers: [
+    {
+      id: "sw360-backend",
+      name: "sw360 backend",
+      type: "oauth",
+      version: "2.0",
+      wellKnown: SW360_API_URL + "/authorization/.well-known/oauth-authorization-server",
+      checks: ["pkce", "state"],
+      idToken: true,
+      // Partial GH Copilot generated- start
+      authorization: {
+        params: {
+          scope: "openid READ WRITE ADMIN", code_challenge_method: "S256", code_challenge: (() => {
+            codeVerifier = codeVerifierGenerator();
+            const codeChallenge = codeChallengeGenerator(codeVerifier);
+            return codeChallenge;
+          })(),
+        }
+      },
+      // Partial GH Copilot generated- end
```

Comment:
Should we remove these comment?

Reply:
Agreed, this should go, and now i raise a warn that we probably need use a snippet scan NOW in the code, as seen like people are using AI generated code, and me as a dinosaur not trust at all on results not literally been copied from elsewhere.

I'll be thinking over how we can automatize snippet scanning.

Reply:
Be very clear and i hope that people understand a little bit better that.

Your Org is not SW360 project, so if your org demands put some kind of disclaimer in the code, the code should be rewritten or removed .
Or we should do a formal inquiry on the community about how we will deal with that.

But again, just because your org want it, not means we should accept.

@KoukiHama @arunazhakesan @ag4ums I'm not taking this lightly, because we are now crossing a thin line.

Reply:
Did the Eclipse Foundation give any insights/ideas/comments on the implementation?

Also, should we use scanoss or something else for snippet scanning?

Reply:
Just came across this article and thought it might be useful: https://www.eclipse.org/projects/guidelines/genai/

Reply:
We should align with the Eclipse Foundation guidelines. There is no reason to put a blanket ban on AI generated code. It should be permitted as per the Eclipse foundation [guidelines](https://www.eclipse.org/projects/guidelines/genai/). 

We should have an organization agnostic view and approach here.

Reply:
@smrutis1 Rebaae, add the disclaimer the way is described by Eclipse guidelines and i'll be ok with that.

Reply:
Done